File: php/rna-transcription/RnaTranscription.php
<?php declare(strict_types=1);

function toRna(string $dna): string
{
    $rna = "";

    foreach (str_split($dna) as $nucleo) {
        $rna .= match ($nucleo) {
            'G' => 'C',
            'C' => 'G',
            'T' => 'A',
            'A' => 'U'
        };
    }

    return $rna;
}

File: python/rna-transcription/rna_transcription.py
def to_rna(dna_strand):
    rna_strand = ""

    for char in list(dna_strand):
        match char:
            case 'G':
                rna_strand += 'C'
            case 'C':
                rna_strand += 'G'
            case 'T':
                rna_strand += 'A'
            case 'A':
                rna_strand += 'U'
                
    return rna_strand
